### ISSUE TITLE
Updated analysis: molecular subtyping for cell lines

### ISSUE THREAD

**Issue by jaclyn-taroni** [AlexsLemonade/OpenPBTA-analysis#509]
#### What analysis module should be updated and why?

We do not currently perform molecular subtyping for cell lines because it requires additional information to map between DNA-seq and RNA-seq samples. This ticket pertains to all molecular subtyping modules, but I only know for certain that it is an issue in the [`molecular-subtyping-HGG`](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/molecular-subtyping-HGG) module.

#### What changes need to be made? Please provide enough detail for another participant to make the update.

Instead of joining using only `Kids_First_Participant_ID` and `sample_id`, we also need to join on `composition` and skip filtering to tumor samples only - for example:

https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/ef178cc6509f7cdb5ca80309d33486eed3aec894/analyses/molecular-subtyping-HGG/02-HGG-molecular-subtyping-subset-files.R#L133 

#### What input data should be used? Which data were used in the version being updated?

Requires the addition of the `composition` column from `pbta-histologies.tsv` for histologies that include cell line samples

#### When do you expect the revised analysis will be completed?

~1 week - anything with identifiers is usually more nuanced than I think!

#### Who will complete the updated analysis?

Likely someone from the CCDL

**Comment by jashapiro** [comment]
One particular complication that arises here is that HGG subtyping (see https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/0edfe5f176105c5c3e8590b678f1d9f735341368/analyses/molecular-subtyping-HGG/results/HGG_defining_lesions.tsv) reclassified samples previously labeled as "Brainstem glioma- Diffuse intrinsic pontine glioma", mostly as "Diffuse midline glioma". This means that the only remaining "Brainstem glioma- Diffuse intrinsic pontine glioma" samples are cell lines. Should they be updated to match, even though they have not been through sub typing and may have odd results when they are?

**Comment by jharenza** [comment]
@jashapiro what do you mean by "odd results when they are"? These should be updated based on the H3 mutations just like the rest of the HGG tumors. Eg: if they have an H3 K28M mutation, they would be deemed diffuse midline glioma with subtype: H3 K28M mutant and if they have the G35 mutation or are wildtype, they would be be deemed "High-grade glioma with subtype H3 G35 or H3 wildtype", such that we would not have anymore "Brainstem glioma- Diffuse intrinsic pontine glioma" in our cohort. The reasoning is that this specific diagnosis was removed from the WHO 2016 classifications and replaced with diffuse midline or high-grade, based on that histone lesion. Does this help?

**Comment by jashapiro** [comment]
@jharenza I guess I don't mean odd results... just that maybe we can't do this with current data? My main point was that the "Brainstem glioma- Diffuse intrinsic pontine glioma" should probably be removed (soon? next data release?) perhaps before this ticket is complete, as it has potential to cause and/or mask issues with analysis that relies on particular sets of disease types. That was the context I ran into, where I wasn't getting DIPG samples in an analysis, but there were still samples labeled that way in the histologies file, so it took me longer than it should have to realize that the label had changed.

**Comment by jharenza** [comment]
@jashapiro sure, I can add no problem - I think we just want to have this subtyping completed. I did not do it before because I think in going through the first PR for tumor subtyping, @jaclyn-taroni and I had talked about handling cell lines separately. @jaclyn-taroni - thoughts on order of operations here?

**Comment by jaclyn-taroni** [comment]
What is the timeline for v15 (#543) @jharenza? I think that will inform my answer.

**Comment by jharenza** [comment]
We can do a v15 release this week if we exclude EPN subtypes (still in progress) and discuss what we want to do with TCGA MAFs (wait on those as well/release our v2 of those while testing WXS params)?

**Comment by jaclyn-taroni** [comment]
Okay, let's wait to make changes to the histologies file until this is completed (and after v15) and we'll consider this a candidate for our next sprint.

**Comment by jaclyn-taroni** [comment]
Because all of these samples are relevant to `molecular-subtyping-HGG`:
```R
> histologies_df %>% 
    filter(composition == "Derived Cell Line") %>% 
    group_by(short_histology) %>% 
    tally()
# A tibble: 1 x 2
  short_histology     n
  <chr>           <int>
1 HGAT               38
```

I think the following point from https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/627#issuecomment-598789232

> We have a standalone BRAF V600E subtype, but this is HGG only, and should be labeled as such (will create an update HGG analysis ticket).

Can be addressed at the same time.

**Comment by jaclyn-taroni** [comment]
I am planning on addressing this issue early this week.

**Comment by jaclyn-taroni** [comment]
Some notes mostly to self:

* In light of https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/609#issuecomment-602821376, I think it makes sense to wait to address the issue of the standalone `BRAF V600E` and more generally add a step/notebook to `molecular-subtyping-HGG` that alters the labels from the `07` notebook (related to: https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/609#issuecomment-602824038). That step would be the last step in `molecular-subtyping-HGG` and incorporate the information from pathologists.
* I am seeing some issues with ambiguous mapping between RNA-seq and DNA-seq samples using the plan I outlined originally and I think it has to do with https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/610

**Comment by jaclyn-taroni** [comment]
Linking to some information for how we can resolve the ambiguous mapping so it's easier to find when this gets addressed: https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/610#issuecomment-655608824

**Comment by jaclyn-taroni** [comment]
When #786 goes in / #754 gets closed, I will follow up with this change.

**Comment by jharenza** [comment]
@kgaonkar6 will pick this up

**Comment by kgaonkar6** [comment]
I made changes to include cell-lines and tried to implement the matching with parent_aliquot_ids along with sample_id in [here](https://github.com/kgaonkar6/OpenPBTA-analysis/blob/update-subtyping-cell-lines-HGG/analyses/molecular-subtyping-HGG/results/HGG_molecular_subtype.tsv)

It seems to only be able to uniquely match duplicate sample_id issue for the following subset 

  | Kids_First_Participant_ID | sample_id | parent_aliquot_id | Kids_First_Biospecimen_ID_DNA | Kids_First_Biospecimen_ID_RNA | molecular_subtype
-- | -- | -- | -- | -- | -- | --
1 | PT_HNZNZ635 | 7316-2151 | 7316-2151-T-409284 | BS_SKRY0BJ4 | BS_49BQS7Z6 | HGG, H3 wildtype
2 | PT_HNZNZ635 | 7316-2151 | 7316-2151-T-409293 | BS_TX8C5VAJ | BS_GXTFW99H | HGG, H3 wildtype
3 | PT_JSFBMK5V | 7316-1769 | 7316-1769-T-366000 | BS_KY1CTGJ3 | BS_1A6MQ9ZA | DMG, H3 K28
4 | PT_JSFBMK5V | 7316-1769 | 7316-1769-T-366002 | BS_RXP2ZRQT | BS_40MP5BWR | DMG, H3 K28
5 | PT_VAJN5QP8 | 7316-2176 | 7316-2176-T-425778 | BS_KSHETTQC | BS_0C7VZC0A | HGG, H3 wildtype
6 | PT_VAJN5QP8 | 7316-2176 | 7316-2176-T-425785 | BS_M659G06J | BS_5GNQC2FF | HGG, H3 wildtype
7 | PT_VTM2STE3 | 7316-2189 | 7316-2189-T-425982 | BS_ERFMPQN3 | BS_PGK832G2 | HGG, H3 wildtype
8 | PT_VTM2STE3 | 7316-2189 | 7316-2189-T-425973 | BS_02YBZSBY | BS_HJRTC9JQ | HGG, H3 wildtype
9 | PT_Y5KY6KN9 | 7316-195 | 7316-195-T-202630 | BS_AYPBAHJ6 | BS_4QW2S1Q8 | DMG, H3 K28
10 | PT_Y5KY6KN9 | 7316-195 | 7316-195-T-203029 | BS_JGKRN7NA | BS_MX23ZY0Y | DMG, H3 K28
11 | PT_Y5KY6KN9 | 7316-195 | 7316-195-T-202952 | BS_XMP9XNR9 | BS_P9JP6JFA | DMG, H3 K28

The Solid-tumor and cell-lines also always have different parent_aliquot_id so can be used to separate out in the matching step. 

However, the following sample_ids for cell-lines share the same parent_aliquot_ids so can't be matched using parental_aliquot_id and might needs additional information (suspension/adhesion ) as mentioned in previous issue tickets. 

[1] "7316-85"   "7316-913"  "7316-3213" "7316-1746" "7316-3058" "7316-3214" "7316-3223" "7316-3230"
 [9] "7316-3217" "7316-3231" "7316-3222" "7316-3227" "7316-3229" "7316-3228" "7316-1763"

**Comment by jharenza** [comment]
@kgaonkar6 let's ask @mkoptyra about this - it may be OK/expected, but not sure.

**Comment by kgaonkar6** [comment]
(Discussion with @jharenza has lead to the following plan for this ticket) 

Step1 We will add a file with the S/A annotation from https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/610#issue-576545712 and use that to match the cell-line DNA and RNA samples in the molecular-subtype-HGG . 

Step2 As part of v18 release we will have S/A information in the pbta-histology file, we will then re-work the molecular-subtype-HGG  module to read that information from the histology file.

**Comment by jharenza** [comment]
closed with #796
